raijin_scripts/deploy/build_environment_module.py

```diff
@@ -158,52 +277,99 @@ def install_pip_packages(pip_conf, variables):
+    """
+    Find default version of the dea module
+
+    :param module_name: Module name
+    :return: Version on success else raise exception
+    """
     cmd = f"module --terse avail {module_name}"
+    LOG.debug("Running command: %s", cmd)
     output = subprocess.run(cmd, shell=True, check=True,
                             stdout=subprocess.PIPE,
                             stderr=subprocess.STDOUT,
+                            universal_newlines=True,
                             encoding='ascii')
     versions = [version for version in output.stdout.splitlines() if f'{module_name}/' in version]
     default_version = [version for version in versions if '(default)' in version]
-    ret_val = None
+
     if default_version:
         ret_val = default_version[0].replace('(default)', '')
     elif len(versions) > 0:
         ret_val = versions[-1]
     else:
         raise Exception('No version of module %s is available.' % module_name)
-
     return ret_val
 
 
-def run_final_commands_on_module(commands, module_name):
+def run_final_commands_on_module(commands, module_path):
+    """
+    Run final commands as per configurations in modulespec.yaml file
+
+    :param commands: List of commands
+    :param module_path: Module path
+    :return: None
+    """
     for command in commands:
-        command = f'module load {module_name}; {command}'
-        run(command)
+        cmd = f'{module_path}/bin/{command}'
+        run_command(cmd)
 
 
 def include_stable_module_dep_versions(config):
+    """
+    Include stable module dependency versions
+
+    :param config: Dictionary of configuration variables
+    :return: None
+    """
     stable_module_deps = config.get('stable_module_deps', [])
     for dep in stable_module_deps:
         default_version = find_default_version(dep)
         dep = dep.replace('-', '_')
         config['variables'][f'fixed_{dep}'] = default_version
 
 
-def main(config_path):
+def reinstall_miniconda(script_name):
+    """
+    Re-install miniconda and npm packages
+
+    :param script_name: Shell script name
+    :return: None
+    """
+    LOG.debug('Re-install miniconda3 before creating new dea-environment module')
+    run_command(f'./{script_name}')
+
+
+def main(config_path, dea_env):
+    """
+    Build new environment module
+
+    :param config_path: Configuration path
+    :param dea_env: Are we building dea-env module?
+    :return: None
+    """
+    assert (dea_env.lower() in ("true", "false")), 'Argument 2 shall be a bool (True or False)'
+
+    # To keep the migration consistency across platforms (macOS/Windows/Linux)
+    ospath = r'%s' % os.getcwd().replace('\\', '/')
+
     logging.basicConfig(level=logging.DEBUG)
+    run_command(f'pip3 install --user pyyaml')
     LOG.debug('Reading config file')
     config = read_config(config_path)
+    variables = config['variables']
+
+    if dea_env.lower() == 'true':
+        reinstall_miniconda('reinstall_miniconda.sh')
```

Comment:
This isn't an acceptable name for this function. It should either be replaced with simply `run_command()`, or it shouldn't take any arguments.

Reply:
`reinstall_miniconda` module deleted and replaced with `run_command()`.